backend/internal/api/api.go

```diff
@@ -40,10 +37,11 @@ func (h Handler) EndpointInitialize(mux *mux.Router) {
 	mux.HandleFunc(h.getAllEndpoint, h.DBGetAllHandler).Methods("GET")
 	mux.HandleFunc(h.getEndpoint, h.DBGetHandler).Methods("GET")
 	mux.HandleFunc(h.postEndpoint, h.DBPostHandler).Methods("POST")
+
 }
 
 // NewHandler returns handler for database manager.
-func NewHandler(dbManager DBManager, idGenerator IdGenerator) Handler {
+func NewHandler(dbManager DBManager, idGenerator utils.IdGenerator) Handler {
```

Comment:
Just a general remark. Current structure of the application seems to be more technical than domain oriented.   
Please, see [this talk](https://www.youtube.com/watch?v=oL6JBUk6tj0&ab_channel=GopherAcademy). Package structure is important so we should give it a thought.